tools/flang2/flang2exe/llutil.cpp

```diff
@@ -2194,6 +2194,86 @@ metadata_args_need_struct(void)
   return ll_feature_metadata_args_struct(&llvm_get_current_module()->ir);
 }
 
+/**
+ * This function returns true for the types supported
+ * in function make_param_op
+ */
+bool
+should_preserve_param(SPTR sptr)
+{
+  bool should_preserve = false;
+  DTYPE dtype = DTYPEG(sptr);
+
+  switch (DTY(dtype)) {
+    case TY_ARRAY:
+    case TY_STRUCT:
+    case TY_BLOG:
+    case TY_SLOG:
+    case TY_LOG:
+    case TY_BINT:
+    case TY_SINT:
+    case TY_INT:
+    case TY_REAL:
+    case TY_INT8:
+    case TY_LOG8:
+    case TY_DBLE:
+    case TY_QUAD:
+    case TY_CMPLX:
+    case TY_DCMPLX:
+    case TY_CHAR:
+      should_preserve = true;
+      break;
+  }
+  return should_preserve;
+}
+
+OPERAND *
+make_param_op(SPTR sptr)
+{
+  OPERAND *oper;
+  DTYPE dtype = DTYPEG(sptr);
+
+  assert(should_preserve_param(sptr), "make_param_op(): unexpected sptr",
```

Comment:
Instead of having another check inside of `should_preserve_param()` you could skip this assert and just emit an error in the `default` case of the following switch statement. This would eliminate the need for `should_preserve_param()` function. Why have the knowledge of "supported types" listed separately over 2 different switch statements? Any changes to one will require a change to the other, which seems like  sub-optimal design.

Reply:
Yes I agree that it looks a little odd. But it has to be done to identify the (now) unsupported cases and let it generate an IR without (out of scope) parameter debug-info in place of producing an error which it would have done without this PR.